src/main/java/uk/gov/laa/ccms/caab/service/ApplicationService.java

```diff
@@ -425,6 +434,62 @@ public AddressFormData getCorrespondenceAddressFormData(final String id) {
         .map(addressFormDataMapper::toAddressFormData).block();
   }
 
+  /**
+   * Retrieves linked cases associated with a specific ID.
+   * This method fetches a list of linked cases, transforms them into a
+   * {@code ResultsDisplay<LinkedCaseResultRowDisplay>} format, and returns the result.
+   *
+   * @param id The unique identifier for the cases to be retrieved.
+   * @return A {@code ResultsDisplay<LinkedCaseResultRowDisplay>} containing the linked cases.
+   */
+  public ResultsDisplay<LinkedCaseResultRowDisplay> getLinkedCases(final String id) {
+    final ResultsDisplay<LinkedCaseResultRowDisplay> results = new ResultsDisplay<>();
+
+    return caabApiClient.getLinkedCases(id)
+        .flatMapMany(Flux::fromIterable) // Convert to Flux<LinkedCase>
+        .map(resultDisplayMapper::toLinkedCaseResultRowDisplay) // Map to ResultRowDisplay
+        .collectList() // Collect into a List
+        .map(list -> {
+          results.setContent(list); // Set the content of ResultsDisplay
+          return results; // Return the populated ResultsDisplay
+        }).block();
```

Comment:
Do we want to be always blocking in our Service tier? I thought we were trying to return Mono's if possible, so the Controller can do multiple async calls etc.

Reply:
Pretty sure it will only be used here, most of the other service methods all block at the end. When it comes to linked cases for the application they are just retrieve through the cascading transaction call. So we wouldn't have to call this method.